account_functions.go

```diff
@@ -0,0 +1,21 @@
+package databricks
+
+import "github.com/databricks/databricks-sdk-go/service/provisioning"
+
+// GetWorkspaceClient returns a WorkspaceClient for the given deployment name.
+// The deployment name for a workspace is present in the deploymentName field of
+// the Workspace struct, which can be obtained by calling w.Workspaces.Get()
+// or w.Workspaces.List().
+//
+// The config used for the workspace is identical to that used for the account,
+// except that the host is set to the workspace host, and the account ID is
+// not set.
+func (c *AccountClient) GetWorkspaceClient(w provisioning.Workspace) (*WorkspaceClient, error) {
+	host := c.Config.Environment().DeploymentURL(w.DeploymentName)
```

Comment:
Can you comment when we have a `DeploymentName` and when we have to use the `AzureResourceID`?

Reply:
Because the deployment name is always present when fetching a workspace, we don't actually have any hard dependency on the AzureResourceID field being set for the core functionality of the Config. In fact, because we are copying the `auth` field, we don't have to preserve any of the fields that are used for configuration. I've simply implemented this because we can compute the AzureResourceId from the Workspace.